serialization/dotnet/text/src/TextSerializationWriter.cs

```diff
@@ -0,0 +1,94 @@
+using System;
+using System.IO;
+using System.Collections.Generic;
+using System.Xml;
+using Microsoft.Kiota.Abstractions;
+using Microsoft.Kiota.Abstractions.Serialization;
+using System.Text;
+using Microsoft.Kiota.Abstractions.Extensions;
+using System.Linq;
+
+namespace Microsoft.Kiota.Serialization.Text;
+
+/// <summary>
+/// The <see cref="ISerializationWriter"/> implementation for text content types.
+/// </summary>
+public class TextSerializationWriter : ISerializationWriter, IDisposable {
+    private readonly MemoryStream _stream = new MemoryStream();
+    private readonly StringBuilder writer = new();
+    private bool written;
+    /// <inheritdoc />
+    public Action<IParsable> OnBeforeObjectSerialization { get; set; }
+    /// <inheritdoc />
+    public Action<IParsable> OnAfterObjectSerialization { get; set; }
+    /// <inheritdoc />
+    public Action<IParsable, ISerializationWriter> OnStartObjectSerialization { get; set; }
+    /// <inheritdoc />
+    public void Dispose()
+    {
+        _stream?.Dispose();
+        GC.SuppressFinalize(this);
+    }
+    /// <inheritdoc />
+    public Stream GetSerializedContent() {
+        var result = writer.ToString();
```

Comment:
To avoid marshalling the string here then having to write it to a stream, we could potentially make the `writer` a `StreamWriter` instead of a `StringBuilder`. 
That way, the `WriteStringValue` function at line 74 (which is eventually called by all the rest) will simple call `writer.Write(value);` then this function(i.e `GetSerializedContent`) would simply flush the writer and return the stream (which will already be written to).
That way we would be essentially writing values directly to the stream instead of writing to the current intermediary which is the `StringBuilder`.

What do you think?

Reply:
good suggestion, fixed

Reply:
Thanks for taking the suggestion